README: Mention additional dependencies when building from Git

b4n opened the pull request:
Fixes #756.

b4n commented:
@elextr please review as the Grand Documentation Master :)

elextr commented on README:
```diff
@@ -47,7 +47,8 @@ $ make
 
 If you are building from a Git clone rather than a source tarball, you need
-to run ./autogen.sh first.
+to run ./autogen.sh first, after having installed GLib's development files.
```
I suggest maybe making it more obvious, you know how nobody reads the README, even less read it carefully, so describing steps that are needed after the steps that follow them is not good.  Instead make two sections, even though they repeat most of it.

Suggest the basic layout:

To make documentation install rst2html, and what happens if its not available

Installation from GIT clone:

Install g* dev (just for my interest why isn't this needed for the tarball?)
autogen.sh (I would suggest only needed first time, but this is the install section not repeated build, so ok)
configure
make
(as root or sudo)
make install

Installing from tarball:

configure
make
(as root or sudo)
make install

b4n commented:
@elextr how's that?

elextr commented on README:
```diff
@@ -40,17 +42,36 @@ GNU versions of these tools are recommended.
+fallback if *rst2html* is missing. When building from Git however, that
+pre-built version is not included and *rst2html* is required by default.
+You can explicitly disable building the user manual using the
+``--disable-html-docs`` *configure* flag, but this will result in not
```
Why doesn't configure automagically disable docs if rst2html is not available?

b4n replied:
because it's a bad thing nobody should want.  at least that's what we currently have decided.

And because we want it for creating tarballs, although we could very well force it just in this case.

elextr commented on README:
```diff
@@ -40,17 +42,36 @@ GNU versions of these tools are recommended.
+and open the online version instead when requested.
+
+From Git clones
++++++++++++++++
```
Building From Git clones

b4n replied:
It's already in the "installation" section, but if you want

elextr commented on README:
```diff
@@ -40,17 +42,36 @@ GNU versions of these tools are recommended.
+
+From a release tarball
+++++++++++++++++++++++
+
```
Installing from a release tarball

elextr commented on README:
```diff
@@ -40,17 +42,36 @@ GNU versions of these tools are recommended.
-If you are building from a Git clone rather than a source tarball, you need
-to run ./autogen.sh first.
+To build the user manual you need *rst2html* from Docutils. A pre-built
+version is available in distribution tarballs and will be used as
```
version of the documentation is ..

elextr commented:
Couple of minor suggestions, otherwise LGTM

elextr commented:
LGTM

eht16 commented:
`the GLib development files **before** following and of the following steps` sounds a bit strange to me. Is there a "following" too much in there or a typo like s/and/any/?

What about `the GLib development files **before** continuing any of the following steps`

eht16 commented:
While we are at the README, I guess we could update the Windows section like this:

```
Windows
-------
Geany runs also under Windows. To download the binary with all necessary
files, visit Geany's homepage. However, the Windows version has a few 
limitations:

- embedded terminal emulation (VTE) is missing
- Windows Vista or newer is required
```
The note for Windows 9x users is not relevant anymore because 9x is soooooo old and not supported at all anyway. Also I dropped the build system note as it got heavily improved in the meantime and works pretty good nowadays.

If you feel like too unrelated changes for this PR, I'd open a new one.

elextr commented:
@eht16 do the installation instructions work on windows, or are there any extras that need mentioning (eg msys2?) or should the section be "Installing on Linux" and a separate section "Installing on Windows" added?

b4n commented:
> `the GLib development files **before** following and of the following steps` sounds a bit strange to me. Is there a "following" too much in there or a typo like s/and/any/?

Both actually, good catch!

I went ahead an chose `the GLib development files **before** running any of the following
commands`, how does that look? (see updated diff, too)
Could use yours or `the GLib development files **before** performing any of the following
steps`, but looks odder to me.

b4n commented:
> While we are at the README, I guess we could update the Windows section like this: […]

I added a commit using your proposal, under your name.  Looks good to me (apart maybe that Windows Vista or newer doesn't seem to fall into the limitation category to me)

eht16 commented:
@b4n `the GLib development files **before** running any of the following commands` reads fine for me. Great.

@b4n thanks for the Windows part.
@codebrainz I chose Vista because at the end of http://www.gtk.org/download/windows.php there is a note that current GTK versions depend on Vista or newer. Maybe the MSYS2 builds have some modifications or it just works by luck.
If you feel we can also mention Windows XP there. On the other hand, both Windows versions are already EOL and so it probably doesn't matter much.

eht16 commented:
> @eht16 do the installation instructions work on windows, or are there any extras that need mentioning (eg msys2?) or should the section be "Installing on Linux" and a separate section "Installing on Windows" added?
Basically, ./configure && make && make install works also on Windows but
only after you have setup the MSYS2 environment. But this is way out of
the scope of the README.
We have detailed instructions about this topic at
http://wiki.geany.org/howtos/win32/msys2.
Since there is also MacOSX which is probably also different, I'd say
let's call the section "Installing on Linux" as you said and maybe drop
a note that building on Windows and MacOSX is more complicated and
covered in the wiki.

elextr commented:
@eht16 yes better to refer complicated things to the wiki.

If the section is "Installing on Linux" then maybe it should first suggest installing from the distro unless you want a newer version?

elextr commented:
Somewhat belatedly remembered that there is an `INSTALL`.

How does that match now?

Should `README` just refer to `INSTALL` ?

eht16 commented:
Here is somewhat restructured README: https://gist.github.com/eht16/882af2eb8ebd1426070dd79138bafc00

Changes:

- add new top-level sections for installing on Windows and MacOSX
- move section "Requirements" into "Installation from source"
- drop the previous Windows section completely as it might not be necessary at all (despite my previous changes to it)
- add a note that building from source on Windows and MacOSX is covered in the wiki

@elextr I don't think referring to `INSTALL` is of much use because it just contains very generic build instructions. It's a stock autotools file and never has been touched, updated or adjusted.
This is probably a new topic on its own.

b4n commented:
I didn't review the proposed changes thoroughly, but maybe you could make it a commit on top of this PR?  Or we merge or drop this one, and you open a new one with the revamped version?

b4n commented:
Well, actually I checked out the diff and it looked mostly fine and small to me, so I went ahead and added it here.  I fixed what i considered being small wording issues, but apart that LGTM.

elextr commented:
LGTM  @b4n @eht16 well done.
